src/common/TableOfContents.ts

```diff
@@ -0,0 +1,12 @@
+export interface HeadingLink {
+  content: string
+  anchorLink: string
+  level: number
```

Comment:
this doesn't seem to be used oddly enough. we might want to remove it

Reply:
Eh sorry I disagree here. The prop may not be used but we want to make sure minimally the props that we need are correct. I think the level is definitely needed because we need to perform indentation on the TOC.

@jacobkwan in the interest of time I think yes it's ok if we don't implement the logic to indent the items in the TOC, but I think it is still useful to have this right now.